### ISSUE TITLE
Height that matches the page size

### ISSUE THREAD

**Issue by ilpiac** [Simbul/baker#677]
Hi,

I'm a graphic designer and I'm new to Baker: I know Baker since old versions, but never used it till now. Now I'm working on a personal project for a magazine.
I read on Baker's homepage on github that "On iPad/iPhone you can constrain the height to match the page size or make it scrollable if you need longer pages".
I'm going to create an article with the main page (title + subheading + small summary) in portrait mode and the main article in landscape mode (text + media) and I would like to have the main page fullscreen in portrait mode.
Does that sentence mean there is a specific way to have a fullscreen page or I have to use CSS as usual to have width & height at 100%?
And can I deploy (locally) a demo app from Xcode on my iPad without an Apple developer licence? Just to know... I can immediately subscribe a licence, obviously, but it would be great to deploy and make some tests on my iPad (not on the simluatro) before buying the dev licence.

Thanks in advance for the info!
Maurizio

**Comment by sportyguy209** [comment]
I'm not a proficient coder or Baker user (just do what I can to get the project done that I'm working on), but I'll give you my newbie answers. I'm sure the big guys will fill in the details.

I believe you may be referring to the scrolling setting in Baker. It is my understanding that you can set it in Baker to scroll a full page at a time or not (where the page will move down as the user swipes).

There may be a way around it, but I found that I had to provision my iPad and iPhone to be able to output from Xcode to my iPad. And, the only way I could do that was to become an official Apple Developer and work through their process.

**Comment by ilpiac** [comment]
@sportyguy209, thanks for your reply!
When I ask if there is a way to "automatically" set page width and height to match the page size (alias, full device width & height) I'm not talking about scrolling the page or not, in a general way. A page scrolls if the content is longer than page height, if content is not so long the page doesn't scroll. But I'm curious to know if there is a way - other than using normal CSS code - to have a page full width & height, especially for background.

For dev subscription: thanks a lot, I will subscribe tomorrow so that I can deploy and make some tests on my iPad.

**Comment by sportyguy209** [comment]
@ilpiac Again, I'm just a newbie who is trying to help if I can.

Just to address a point that others may not get, yes, if you have more content than will fit on a page, the page will automatically scroll. But, as I understand it, through Baker, you can set that method (scroll) to jump full page by full page or to scroll through your content, as necessary.

Sorry, I'm not really understanding your question. If you don't specify in CSS your page width, your content will fill the size of the screen. The same can be said for height. If your content is greater than the screen can hold it will expand as needed. If you are just trying to limit your content to the size of the screen, through CSS, you can lock in your page width and height. What are you trying to accomplish?

**Comment by ilpiac** [comment]
@sportyguy209, thanks for the info, I missed that through Baker you can choose between "jump to next page" and "scrolling": it could be really useful.

About the other question: I was just wondering if the phrase "On iPad/iPhone you can constrain the height to match the page size or make it scrollable if you need longer pages" you can read on Baker github page means something particular or exactly what you have written :) That is you have to use CSS to set width and height of your page and make it a fullpage if and when your content doesn't scroll (for instance on a cover page).

**Comment by folletto** [comment]
Thanks for asking. :)
Yes, you got it correctly: it's just standard CSS, nothing else. We try to keep Baker as "pure" HTML as possible. :)

**Comment by ilpiac** [comment]
I have a kind of cover page (portrait mode) with a fullpage background, then you rotate your iPad to read the real article in landscape mode. But since everything is inside the same structure (the cover is made by header+hgroup tags, the article is inside an article tag) it's no so easy to have a height 100%, since the real content is longer than the cover page.
So, I hoped in some "tricky code" to make things easier ;)
But I really understand that things has to be as basic and pure as possible, this is why I design every article in a semantic way: I want a really easy HTML5 structure that you can change in different ways with CSS.
So, I will try and find a solution with CSS :)

**Comment by folletto** [comment]
In CSS you have media-queries that will allow you to basically even redesign entirely the page. Check them out. :)

**Comment by ilpiac** [comment]
I actually use them to show/hide different contents in landscape and portrait mode.
The fullpage cover is just a small part of the actual HTML page: trying to set height at 100% (with html 100% + body 100% etc) doesn't work but this is not a problem, I will find a way with CSS for sure!
A "tricky code" directly in Baker would have been quickier for me, but not so correct, I know :)

**Comment by ilpiac** [comment]
Just found a kind of solution: I've set html and body height to 100% and then the wrapper DIV to min-height 100% for portrait. This to have a colored background fullpage. Then I set every single element inside the wrapper DIV so that they are horizontally centered and "visually" almost vertically centered. Not a real center position, but it's fine for me :)

The problem is that I have a quite full/complex page. Main idea: a "core content" made with HTML5 with support fore more than a single language. Different languages supported with sessionStorage and every language wrapped in a different DIV. ARTICLE put inside every DIV, with different contents visible in portrait and landscape mode: for example text in portrait and media in landscape. Or cover page in portrait mode and article (text + media) in landscape. Basically the HTML5 structure has to be semantically correct (article, section, aside tags and not div or span if they don't mean anything), I want ot use it as the starting point also for web and not only with Baker.

Anyway, Baker is really a great great project!!! I love it. I'm a graphic designer, mainly for print, I know HTML5 and CSS (but I'm not really a pro dev) but without Baker it would be really impossible for me to develop anything :)
